provider/resource_collection_test.go

```diff
@@ -37,3 +41,32 @@ func TestResourceCollectionPlan(t *testing.T) {
 		},
 	})
 }
+
+func TestAccResourceCollectionLifecycle(t *testing.T) {
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		CheckDestroy: testAccCheckResourceDestroy(resourceNameCollection, testAccCollectionLifecycle_collectionName),
+		Steps: []resource.TestStep{
+			{
+				ProtoV6ProviderFactories: providerFactories,
+				ConfigDirectory:          config.TestStepDirectory(),
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttr("itsi_splunk_collection.test", "name", testAccCollectionLifecycle_collectionName),
+					testAccCheckResourceExists(resourceNameCollection, testAccCollectionLifecycle_collectionName),
+				),
+			},
+			{
+				ProtoV6ProviderFactories: providerFactories,
```

Comment:
Just a question, where do you destroy resources on lifecycle? :) 
This test is about initial apply and clean plan after, but if thing is created/applied first time, all subsequent tests should have 2 times just empty plan, right?

Reply:
Destroy runs automatically after the last TestStep of a TestCase finishes. After that the testing framework runs the CheckDestroy function to validate that the resource has actually been destroyed.
So ultimately this test case does the following:
1) create collection data 
2) update collection data 
3) run plan & refresh after the update, and verify that there's no diff in the plan
4) destroy collection data